Integrate model into app UI

angusYuhao opened the pull request:
1. Connected `runModel()` function with app UI
2. Graph and plot results from model

liu-allan commented:
Application says it is not responding when running the model. Is this normal?

![image](https://user-images.githubusercontent.com/44624435/229968521-ea01a33f-f33e-4f9f-a7fd-52c615f6ecba.png)

liu-allan commented on widget_pages/dashboard.py:
```diff
@@ -202,5 +280,24 @@ def showPatientListWindow(self):
+
+    def runMatLabModel(self, patient, num_cycles):
+        bsa = float(patient.bsa)
+        num_cycles = float(num_cycles + 1)
```
why are we running for `num_cycles + 1`?

angusYuhao replied:
We are assuming that when the user enters 1 cycle to run, they want to run 1 cycle into the future. That means the model needs to run both the current cycle (for reference trajectory) and the future cycles -> 2 cycles to run

liu-allan replied:
Actually, I have more questions. If the patient has 3 measurements to use and you specify 2 cycles, does it calculate 5 cycles total? (3 for historical then 2 for the future?)

angusYuhao replied:
no, it should only run 3 cycles, the current one and 2 into the future

liu-allan replied:
So it only runs using the most recent measurement then?

angusYuhao replied:
yeah, I believe so

liu-allan reviewed (commented):
Perhaps we should have a confirmation box appear when the user wants to change pages. It takes a long time to load and I think we should make sure the user doesn't accidentally switch pages

liu-allan reviewed (commented):
On line 73 in `dashboard.py` could you change the axis Neutrophil Count title to `Neutrophil Count (# Cells/L) x 1e9`.

ianyechen commented on matlab_script.py:
```diff
@@ -24,5 +24,26 @@ def runModel(bsa: float, numCycles: int, dosage: list[float], ANC_measurements:
+    time = [time[i][0] for i in indices]
+    reactive_trajectory = [reactive_trajectory[i][0] for i in indices]
+    anticipatory_trajectory = [anticipatory_trajectory[i][0] for i in indices]
+    reactive_dosage = [reactive_dosage[i][0] for i in indices]
```
@liu-allan

liu-allan replied:
i can change this in the matlab

ianyechen reviewed (commented):
![image](https://user-images.githubusercontent.com/44624592/230012566-6972a074-24d0-4bcc-9daa-dc4e3efeb05d.png)
Not a big deal, but should we move the error message for the "number of calculation cycles" field being empty to like its actual box?

ianyechen commented on matlab_script.py:
```diff
@@ -24,5 +24,13 @@ def runModel(bsa: float, numCycles: int, dosage: list[float], ANC_measurements:
     eng.quit()
 
-    return time, nominal_trajectory, linearized_trajectory, reactive_trajectory, anticipatory_trajectory, reactive_dosage, anticipatory_dosage
+    # print("time: ", len(time), ", reactive_anc: ", len(reactive_trajectory), ", reactive_dosage: ", len(reactive_dosage), ", anticipatory_anc: ", len(anticipatory_trajectory), ", anticipatory_dosage: ", len(anticipatory_dosage))
```
Don't forget to remove print statements when you're done :D

liu-allan reviewed (dismissed):
Looks good, just remove the prints

angusYuhao commented:
Should have addressed all comments and concerns except for the waiting screen when you press calculate, once you guys have reviewed it we can merge this. I will open a new PR for the waiting screen if we have time

ianyechen commented:
Another minor comment:

![image](https://user-images.githubusercontent.com/44624592/230274160-e64ea607-144f-42a0-bf0b-60d2bc99cf8d.png)
What do you guys think about just completely removing the "Cancel" button. Right now it's in the same box as the calculate and it's kinda misleading. We have the toolbar on the left already... maybe that's enough? @angusYuhao @liu-allan @xieanne 

(This comment does not block the PR tho)

ianyechen commented:
Also, have you decided on what to do in terms of saving the calculated data in memory yet? I think it would be nice to not have the model calculate again if nothing has changed? 

I guess it's not a big MUST to do. But would be nice.

ianyechen commented:
![image](https://user-images.githubusercontent.com/44624592/230274833-4eb2a2fc-ed86-4ebc-adcf-60f80a73eac3.png)

When there's no historical data, and you do 1 calculation cycle. No output is expected right? Should we have some kind of prompt that tells the user this? Or maybe just stop them from calculating if there's no historical data?

ianyechen reviewed (approved):
LGTM, left some unblocking comments though.

xieanne reviewed (approved):
LGTM
